students/pooria_k/lesson01/test.py

```diff
@@ -0,0 +1,146 @@
+        self.assertEqual(self.furniture_1_dic['description'], 'sofa')
+        self.assertEqual(self.furniture_1_dic['market_price'], '1200')
+        self.assertEqual(self.furniture_1_dic['rental_price'], '100')
+        self.assertEqual(self.furniture_1_dic['material'], 'leather')
+        self.assertEqual(self.furniture_1_dic['size'], '22x33x44')
+
+
+
+class MainModuleTest_addnew_item(TestCase):
+
+    def test_addnew_furniture(self):
+        input_furniture = ('444', 'bed', 150, 'y', 'wood', 'L')
+        input_app = ('555', 'laptop', 50, 'n', 'y', 'samsung', 120)
+        input_inventory = ('666', 'test', 5, 'n', 'n')
+        expected_item_dic = {'444': {'description': 'bed',
+                                     'market_price': 180.0,
+                                     'product_code': '444',
+                                     'rental_price': 150,
+                                     'material': 'wood',
+                                     'size': 'L'},
+                             '555': {'description': 'laptop',
+                                     'market_price': 120.0,
+                                     'product_code': '555',
+                                     'rental_price': 50,
+                                     'brand': 'samsung',
+                                     'voltage': 120},
+                             '666': {'description': 'test',
+                                     'market_price': 10.0,
+                                     'product_code': '666',
+                                     'rental_price': 5
+                                                    }}
+
+
+        with patch('builtins.input', side_effect=input_furniture):
+            main.FULL_INVENTORY = {}
+            with patch('inventory_management.market_prices.get_latest_price', return_value=180.0):
+
+                main.addnew_item()
+                self.assertDictEqual(main.FULL_INVENTORY['444'], expected_item_dic['444'] )
+
+        with patch('builtins.input', side_effect=input_app):
+            main.FULL_INVENTORY = {}
+            with patch('inventory_management.market_prices.get_latest_price', return_value=120.0):
+
+                main.addnew_item()
+                self.assertDictEqual(main.FULL_INVENTORY['555'], expected_item_dic['555'] )
+
+        with patch('builtins.input', side_effect = input_inventory):
+            with patch('inventory_management.market_prices.get_latest_price', return_value=10.0):
+
+                main.addnew_item()
+                self.assertEqual(main.FULL_INVENTORY['666'], expected_item_dic['666'])
+
+
+    def test_exit(self):
+        with self.assertRaises(SystemExit):
+            main.exit_program()
+
+    def test_item_info(self):
+
+        with patch ('builtins.input', side_effect = '444'):
+            main.FULL_INVENTORY={}
+            self.assertEqual(main.item_info(),  print("Item not found in inventory"))
+
+        with patch ('builtins.input', side_effect = '444'):
```

Comment:
This line should be: 

`with patch ('builtins.input', side_effect = ['444']):`

Note the additional square brackets ([]). _side_effect_ is treated as an iterable, if you just leave the string, each time you call _input_ from _main.item_info()_ it will really pull one character (a '4', in your case). 

This should bring coverage for main.py to 90%, which meets the requirement. There are ways to test the main execution block of the file, but that's usually not done.

Regards,

Luis